internal/controller/utils/sync/nfspvc.go

```diff
@@ -182,3 +184,54 @@ func preparePv(nfspvc danaiov1alpha1.NfsPvc) corev1.PersistentVolume {
 	}
 	return pv
 }
+
+// isConnectedPVCDeleted returns if the connected PVC is deleted
+// connected PVC considered deleted when the PV phase is Released or Failed,
+// or the PV phase in the nfspvc is bound, the PVC phase in the nfspvc is pending and the UID in the PV claimRef is different from the PVC UID.
+func isConnectedPVCDeleted(ctx context.Context, k8sClient client.Client, PV corev1.PersistentVolume, nfspvc danaiov1alpha1.NfsPvc) (bool, error) {
+	if isPVReleased(PV) {
+		return true, nil
+	}
+	if isPVFailed(PV) {
+		return true, nil
+	}
+	if isPVCInRecreation, err := isPVCInRecreationState(ctx, k8sClient, nfspvc, PV); err != nil {
+		return false, err
+	} else {
+		return isPVCInRecreation, nil
+	}
+}
+
+// isPVReleased returns true if the PV phase is Released.
+func isPVReleased(PV corev1.PersistentVolume) bool {
+	return (PV.Status.Phase == corev1.VolumeReleased)
+}
+
+// isPVFailed returns true if the PV phase is Failed.
+func isPVFailed(PV corev1.PersistentVolume) bool {
+	return (PV.Status.Phase == corev1.VolumeFailed)
+}
+
+// isPVCInRecreationState returns true if the PVC in recreation state(when the PV phase in the nfspvc is bound, the PVC phase in the nfspvc is pending and the UID in the PV claimRef is different from the PVC UID).
```

Comment:
Need to fix the comment here to:

// isPVCInRecreationState returns true if the PVC in recreation state, i.e. when
// the PV phase of the nfspvc is bound,  and
// the PVC phase in the nfspvc is pending, and 
// the UID in the PV claimRef is different from the PVC UID.